### ISSUE TITLE
Bug in APE-14 description of WCS

### ISSUE THREAD

**Issue by astrofrog** [spacetelescope/gwcs#495]
I'm running into issues with a solar GWCS where the high-level objects are astropy coordinates with default values of arcseconds. To reproduce this you will need the following dataset:

https://github.com/astropy/reproject/blob/main/reproject/tests/data/aia_171_level1.asdf

The issue:

```python
In [1]: import asdf

In [3]: aia = asdf.open("/Users/tom/Code/Astropy/reproject/reproject/tests/data/aia_171_level1.asdf")

In [5]: aia['wcs'].world_axis_object_components
Out[5]: [('celestial', 0, 'spherical.lon'), ('celestial', 1, 'spherical.lat')]

In [7]: aia['wcs'].world_axis_units
Out[7]: ('deg', 'deg')

In [8]: coord = aia['wcs'].pixel_to_world(1, 2)

In [9]: coord.spherical.lon
Out[9]: <Longitude -1203.0967792 arcsec>
```

Basically the issue is that ``world_axis_object_components`` is incorrect because if applied to the high level objects, it will not give values that could be passed directly to ``world_to_pixel_values`` (which it should). I *think* the following would be more correct:

```python
In [5]: aia['wcs'].world_axis_object_components
Out[5]: [('celestial', 0, 'spherical.lon.deg'), ('celestial', 1, 'spherical.lat.deg')]
```

Is this an issue in GWCS, or in the ASDF file? (I don't understand where all the layers live).

**Comment by nden** [comment]
I'm not well versed in the API. Which method gives a wrong result and what do you expect to get? 
Also how would one apply the ``world_axis_object_components`` function  to high level objects?

For a spherical system the method is defined [here](https://github.com/spacetelescope/gwcs/blob/master/gwcs/coordinate_frames.py#L357)  
Ping @Cadair as he contributed to this.

**Comment by astrofrog** [comment]
@nden the main thing that needs to be done is that ``world_axis_object_components`` should be more specific about what units to get the values in, e.g.

```
In [5]: aia['wcs'].world_axis_object_components
Out[5]: [('celestial', 0, 'spherical.lon.deg'), ('celestial', 1, 'spherical.lat.deg')]
```

or it could be that ``world_axis_units`` should be ``arcsec`` instead of ``deg``. I'll let @Cadair chime in though.

**Comment by Cadair** [comment]
I think that it's world_axis_units that's wrong. We should support things in native arcsec.

I am not working today, but will try and look next week.

**Comment by Cadair** [comment]
ok, I have tracked this down.

In the asdf file included in the reproject test suite some muppet (me), didn't set the units of the `CelestialFrame` object to `u.arcsec` which I should have done.

However, this brings up an interesting point, it appears to be true that the units of `CelestialFrame` should match the default units of the SphericalRepresentation on the `reference_frame` (in this case [arcsec](https://github.com/sunpy/sunpy/blob/main/sunpy/coordinates/frames.py#L500-L502)), so why don't we automatically default to that?

**Comment by Cadair** [comment]
I have now fixed this in #512 and also #457
